core/client/src/main/java/alluxio/client/block/BlockWorkerClient.java

```diff
@@ -120,14 +120,16 @@
    * locks and temporary files.
    *
    * @throws IOException if an I/O error occurs
+   * @throws InterruptedException if this thread is interrupted
    */
-  void sessionHeartbeat() throws IOException;
+  void sessionHeartbeat() throws IOException, InterruptedException;
```

Comment:
Without this, it will throw runtime exception in DynamicResourcePool. 

With this InterruptedException, this is the flow
mHeartbeatFuture.cancel() => DynamicResourcePool#acquire() throws InterruptedException => sessionHeartbeat throws InterruptedException => InterruptedException is handled by HeartbeatThread#run

Without this, this is the flow:
mHeartbeatFuture.cancel() => DynamicResourcePool#acquire() throws RuntimeException => sessionHeartbeat logs the exception
Then in the log, we see lots of runtimeException traces